### DISCUSSION TITLE
How to embed a beam element in 3d solid elements?

### DISCUSSION THREAD

**Discussion by stonebeta** [idaholab/moose#28680] [Q&A Meshing]
I am trying to embed fibers (modeled with 1D beam elements later) in the matrix (3d solid elements), like this in Gmsh

![a](https://github.com/user-attachments/assets/dc3a5197-df5c-44eb-8363-6beeb0fa812e)

I have reviewed similar discussions #24887 and #24174, but I feel that they do not address this issue.

I can generate 1D mesh and 3d mesh separately in Gmsh. For example
```
SetFactory("OpenCASCADE");
//+
Box(1) = {0, 0, 0, 1, 1, 1};
//+
Point(9) = {0.2, 0.2, 0.2, 1.0};
//+
Point(10) = {0.8, 0.8, 0.8, 1.0};
//+
Line(13) = {9, 10};
//+
Curve{13} In Volume{1};
//+
Physical Curve("fiber", 21) = {13};
Mesh.SaveAll = 0; 

Mesh 3;
Save "fibers.msh";

Physical Surface("left", 13) = {1};
//+
Physical Surface("right", 14) = {2};
//+
Physical Surface("top", 15) = {4};
//+
Physical Surface("bottom", 16) = {3};
//+
Physical Surface("front", 17) = {6};
//+
Physical Surface("back", 18) = {5};
//+
Physical Volume("block", 20) = {1};
Mesh.SaveAll = 1; 

Save "matrix.msh";
```

However, the 1D boundary "fiber" in the 3D mesh is not loaded using `FileMeshGenerator`. 

So it seems I cannot use `StitchedMeshGenerator` either, because I cannot find the corresponding boundary positions between the two meshes.

Could you provide any suggestions or hints? Thank you.

**Comment by GiudGiud** [comment]
Hello

The Gmsh reader must not be supporting these mixed dimensional mesh if it is not working. 
How did you check it is not loaded? What does the mesh summary show in the `--mesh-only` mode?

The `StitchedMeshGenerator` is a good idea. For that to work you will need to:
- load the two meshes in two FileMeshGenerator
- create nodeset(s) for the ends of the 1D mesh, both in the 1D mesh and the 3D mesh
- use the nodesets for stitiching

**Comment by jmeier** [comment]
Dear @xiekai-mc ,

The MSH file format does not differentiate between "sidesets" and "elements of lower dimensionality". Your beam is embedded in volume elements - this is here considered as "element of lower dimensionality". So Moose needs a hint, if those lines are sidesets (default) or elements. 

You can do that by (1) appending "lower_dimensional_block" to the name of the beam or (2) after the name (the second option looks nicer, but you have to edit the msh file):

```
$MeshFormat
4.1 0 8
$EndMeshFormat
$PhysicalNames
...
0 1 "PointA"                                          ← sideset (node)
1 2 "topleftedge"                                     ← sideset (line)
1 3 "lineSignPole000_lower_dimensional_block"         ← block with beam-elements
1 4 "lineSignPole001" lower_dimensional_block         ← block with beam-elements
2 5 "BoundaryZMin"                                    ← sideset (surface)
2 6 "Sign" lower_dimensional_block                    ← block with shell-elements
...
```

(Do not copy the `← block with beam-elements`. This is just for illustration.)
